nixos/modules/services/backup/restic.nix

```diff
@@ -134,25 +162,41 @@ in
         let
           extraOptions = concatMapStrings (arg: " -o ${arg}") backup.extraOptions;
           resticCmd = "${pkgs.restic}/bin/restic${extraOptions}";
+          filesFromTmpFile = "/run/restic-backups-${name}/includes";
+          backupPaths = if (backup.dynamicFilesFrom == null)
+                        then concatStringsSep " " backup.paths
+                        else "--files-from ${filesFromTmpFile}";
+          pruneCmd = optional (builtins.length backup.pruneOpts > 0) [
+            ( resticCmd + " forget --prune " + (concatStringsSep " " backup.pruneOpts) )
+            ( resticCmd + " check" )
```

Comment:
I'm not 100% sure but I think check is not needed when running prune.
Check https://forum.restic.net/t/should-i-run-restic-check/1227

but it seems some people might want to run check --read-data from time to time.

Reply:
Also the prune command doesn't run because the check one is on the same line in the final systemd service file:
![image](https://user-images.githubusercontent.com/24027/73490945-8d5de200-437b-11ea-9943-1ff5559bfccf.png)

Reply:
Yeah, I'm seeing that missing newline too.  I don't understand why, but I see it.

Reply:
Also I don't think I'm quite ready to rely on the accepted solution in that forum thread, because of the quote from the docs later on: https://forum.restic.net/t/should-i-run-restic-check/1227/9

Reply:
found the problem with the missing newline, see https://github.com/jerith666/nixpkgs/commit/dc6616b190e0aa6670324ad5078b365dd39fd402

Reply:
> Also I don't think I'm quite ready to rely on the accepted solution in that forum thread, because of the quote from the docs later on: https://forum.restic.net/t/should-i-run-restic-check/1227/9

Yeah that's weird. fd0 is the main dev so we are supposed to be able to trust him. But he didn't bother to reply on the last comment.

Maybe we should ask him. It would be a shame to run an expansive operation like `check` daily until the next release.

Reply:
My test passes with https://github.com/jerith666/nixpkgs/commit/dc6616b190e0aa6670324ad5078b365dd39fd402 . 

Do you think it would be a good idea to make running `restic check` optional for now? And figure out the true best practices from the upstream dev later.